File: December - 17/SakthiDharan01_racing_turtles.py
import sys

def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    it = iter(data)
    target = int(next(it))
    n = int(next(it))
    positions = [int(next(it)) for _ in range(n)]
    speeds = [int(next(it)) for _ in range(n)]
    if n == 0:
        print("No turtle fleets formed.")
        return
    turtles = sorted(zip(positions, speeds), reverse=True)
    fleets = 0
    slow_time = -1.0
    for pos, speed in turtles:
        time = (target - pos) / speed
        if time > slow_time:
            fleets += 1
            slow_time = time
    print(f"The number of turtle fleets is: {fleets}")

if __name__ == "__main__":
    solve()

File: December - 17/SakthiDharan01_racing_turtles.java
import java.util.*;

public class SakthiDharan01_racing_turtles {
    private static class Turtle {
        int pos, speed;
        Turtle(int p,int s){pos=p;speed=s;}
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;
        int target = sc.nextInt();
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        int[] pos = new int[n];
        int[] speed = new int[n];
        for (int i=0;i<n && sc.hasNextInt();i++) pos[i]=sc.nextInt();
        for (int i=0;i<n && sc.hasNextInt();i++) speed[i]=sc.nextInt();
        if (n==0){
            System.out.println("No turtle fleets formed.");
            return;
        }
        Turtle[] arr = new Turtle[n];
        for (int i=0;i<n;i++) arr[i]=new Turtle(pos[i], speed[i]);
        Arrays.sort(arr, (a,b)->b.pos-a.pos);
        int fleets=0;
        double slow=-1;
        for (Turtle t: arr){
            double time = (double)(target - t.pos) / t.speed;
            if (time > slow){
                fleets++;
                slow = time;
            }
        }
        System.out.println("The number of turtle fleets is: " + fleets);
        sc.close();
    }
}
